GrandCentralBoard/Utilities/GoogleAPIDataProvider.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  GoogleAPIDataProvider.swift
+//  GrandCentralBoard
+//
+//  Created by Michał Laskowski on 06.04.2016.
+//  Copyright © 2016 Oktawian Chojnacki. All rights reserved.
+//
+
+import Alamofire
+import Operations
+
+enum APIDataError : ErrorType {
+    case AuthorizationError
+    case UnderlyingError(NSError)
+}
+
+final class GoogleAPIDataProvider {
+
+    private let tokenProvider: OAuthTokenProvider
+    private var accessToken: AccessToken?
+
+    private let networkRequestManager: NetworkRequestManager
+
+    private let operationQueue = OperationQueue()
+
+    init(tokenProvider: OAuthTokenProvider, networkRequestManager: NetworkRequestManager = Manager()) {
+        self.tokenProvider = tokenProvider
+        self.networkRequestManager = networkRequestManager
+    }
+
+    private func refreshTokenOperation() -> Operation {
+        let refreshTokenOperation = BlockOperation(block: { [weak self] (continueWithError) in
+            guard let strongSelf = self else {
+                continueWithError(error: nil)
+                return
+            }
+
+            strongSelf.tokenProvider.accessTokenFromRefreshToken({ result in
+                switch result {
+                case .Failure(let error):
+                    continueWithError(error: error)
+                case .Success(let value):
+                    self?.accessToken = value
```

Comment:
`strongSelf` would be referenced from another block, so it could retain it. I think actually `strongSelf` is not needed here, because if the object was deallocated it would actually make no difference if `accessTokenFromRefreshToken` was executed.